server/images.go

```diff
@@ -72,46 +66,78 @@ type Model struct {
 	Template *template.Template
 }
 
-// CheckCapabilities checks if the model has the specified capabilities returning an error describing
-// any missing or unknown capabilities
-func (m *Model) CheckCapabilities(caps ...Capability) error {
-	var errs []error
-	for _, cap := range caps {
-		switch cap {
-		case CapabilityCompletion:
-			r, err := os.Open(m.ModelPath)
-			if err != nil {
-				slog.Error("couldn't open model file", "error", err)
-				continue
-			}
-			defer r.Close()
+// Capabilities returns the capabilities that the model supports
+func (m *Model) Capabilities() []model.Capability {
+	capabilities := []model.Capability{}
 
-			// TODO(mxyng): decode the GGML into model to avoid doing this multiple times
-			f, _, err := ggml.Decode(r, 0)
-			if err != nil {
-				slog.Error("couldn't decode ggml", "error", err)
-				continue
-			}
+	// Check for completion capability
+	r, err := os.Open(m.ModelPath)
+	if err == nil {
+		defer r.Close()
 
+		// TODO(mxyng): decode the GGML into model to avoid doing this multiple times
```

Comment:
I think the comment can go. The problem before was it was decoding the ggml inside of the for loop which you fixed.